tox.ini

```diff
@@ -1,6 +1,22 @@
 [tox]
-envlist = py26, py27, py33, py34, py35, pypy
+envlist = py{27,33,34,35,py}-requests{,2.1.0,2.8.1,2.9.1},codestyle
 
 [testenv]
+deps =
+    pylint
+    pytest
+    pytest-gitignore
+    pytest-mock
+    pytest-pylint
+
+    py{26,27,33,34,35,py}-requests: requests
+    py{26,27,33,34,35,py}-requests2.1.0: requests==2.1.0
+    py{26,27,33,34,35,py}-requests2.8.1: requests==2.8.1
+    py{26,27,33,34,35,py}-requests2.9.1: requests==2.9.1
+
 commands = py.test
-deps = -rrequirements.txt
+
+[testenv:codestyle]
+deps = pycodestyle
```

Comment:
Good question. I don't know much about pylint to be able to provide a complete answer though.
In my experience I find that pylint and pycodestyle can often find different things.
While pylint often lists harder to find errors like bad variable names pycodestyle focuses on PEP-8 compliance.